src/writeData/components/ExecuteCodeBlock.tsx

```diff
@@ -0,0 +1,85 @@
+// Libraries
+import React, {FC, createContext, useState, useEffect, useContext} from 'react'
+import {format_from_js_file} from '@influxdata/flux'
+
+// Utils
+import {parse} from 'src/external/parser'
+import {getBucketsFromAST, updateBucketInAST} from 'src/flows/context/query'
+import {WriteDataDetailsContext} from 'src/writeData/components/WriteDataDetailsContext'
+
+// Constants
+import {CLIENT_DEFINITIONS} from 'src/writeData'
+
+interface ExecuteCodeBlockContextType {
+  query?: string
+  changeQuery: (query: string) => void
+  executeCodeBlock: string
+}
+
+export const EXECUTE_CODE_BLOCK_CONTEXT: ExecuteCodeBlockContextType = {
+  query: null,
+  changeQuery: () => {},
+  executeCodeBlock: '',
+}
+
+export const ExecuteCodeBlockContext = createContext<
+  ExecuteCodeBlockContextType
+>(EXECUTE_CODE_BLOCK_CONTEXT)
+
+interface Props {
+  query?: string
+  contentID: string
+}
+
+const getBucketFromQuery = (query: string): string => {
+  const ast = parse(query)
+
+  return getBucketsFromAST(ast)[0]
+}
+
+const QUERY_PLACEHOLDER = '<%= query %>'
+
+const ExecuteCodeBlockProvider: FC<Props> = ({query, contentID, children}) => {
+  const def = CLIENT_DEFINITIONS[contentID]
+  const [currentQuery, setQuery] = useState(query ?? def.query)
+  const [executeCodeBlock, setExecuteCodeBlock] = useState('')
+  const {bucket} = useContext(WriteDataDetailsContext)
+  const [selectedBucket, setSelectedBucket] = useState(
+    query ? getBucketFromQuery(query) : bucket?.name
+  )
+
+  useEffect(() => {
+    if (bucket && selectedBucket !== bucket?.name) {
+      setSelectedBucket(bucket?.name)
+    }
+  }, [bucket, selectedBucket, currentQuery, bucket?.name, query])
+
+  useEffect(() => {
+    const executeCode = def.execute.toString()
+    if (!query) {
+      setExecuteCodeBlock(executeCode.replace(QUERY_PLACEHOLDER, def.query))
```

Comment:
Use the template context's transform function and not replace. We want one definition of how a template is parsed / variables are replaced in it, not many. Let's say there's a conflict on <%= and we have to change it to <[[ or else the whole internet falls over. We want that definition to be centralized so that we don't have some case still using <%= somewhere that evaded our best grep usage